analyzers/nodejs/nodejs.go

```diff
@@ -117,48 +117,30 @@ func Discover(dir string, options map[string]interface{}) ([]module.Module, erro
 			return filepath.SkipDir
 		}
 
-		if !info.IsDir() && info.Name() == "package.json" {
-			name := filepath.Base(filepath.Dir(path))
-			// Parse from project name from `package.json` if possible
-			var manifest npm.Manifest
-			if manifest, err = npm.FromManifest(path, "package.json"); err == nil && manifest.Name != "" {
-				name = manifest.Name
-			}
+		if info.IsDir() {
+			return nil
+		}
 
-			log.Debugf("Found NodeJS project: %s (%s)", path, name)
-			path, err = filepath.Rel(dir, path)
-			if err != nil {
-				panic(err)
-			}
-			modules = append(modules, module.Module{
-				Name:        name,
-				Type:        pkg.NodeJS,
-				BuildTarget: filepath.Dir(path),
-				Dir:         filepath.Dir(path),
-			})
-		} else if !info.IsDir() {
-			for _, filename := range npm.PossibleLockfileFilenames {
-				if info.Name() == filename {
-					name := filepath.Base(filepath.Dir(path))
-					// Parse from project name from `package-lock.json` or `npm-shrinkwrap.json` if possible
-					var lockfile npm.Lockfile
-					if lockfile, err = npm.FindAndReadLockfile(path); err == nil && lockfile.Name != "" {
-						name = lockfile.Name
-					}
-
-					log.Debugf("Found NodeJS project: %s (%s)", path, name)
-					path, err = filepath.Rel(dir, path)
-					if err != nil {
-						panic(err)
-					}
-					modules = append(modules, module.Module{
-						Name:        name,
-						Type:        pkg.NodeJS,
-						BuildTarget: filepath.Dir(path),
-						Dir:         filepath.Dir(path),
-					})
-					break
+		filesToCheckFor := append([]string{"package.json"}, npm.PossibleLockfileFilenames...)
```

Comment:
Let's add `yarn.lock` here too. I love the short refactor, this cleaned up discovery a lot.